Source/Patches.cs

```diff
@@ -522,6 +571,459 @@ public static void Postfix()
 		}
 	}
 
+
+	// send rooms data from time to time
+	//
+	[HarmonyPatch(typeof(UIRoot), nameof(UIRoot.UIRootUpdate))]
+	public static class UIRoot_UIRootUpdate_Rooms_Update
+	{
+		static int roomsUpdateTicks = -1;
+
+		public static void Postfix()
```

Comment:
Thanks, My view is same as yours here.  I was not aware it would be many.  I'm new to modding rimworld, so was not aware.  I think this should be revised..
What I want to do is:
list the rooms so ai is aware of rooms (not single door rooms though) and the general brightness of the rooms.  I'm not sure how to do this though.  I figured using a single cell would balance performance, code complexity and be used as a heuristic.  Please advise on a better way

Reply:
This should avoid rooms that are just doors, as it only counts rooms that have a proper name, it excludes "room" and "none" for example.